src/test/java/ast/ASTNodesKtTest.kt

```diff
@@ -0,0 +1,173 @@
+package ast
+
+import antlr.TreepatVisitorImplementation
+import antlr.generate.TreepatLexer
+import antlr.generate.TreepatParser
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+import org.antlr.v4.runtime.CharStreams
+import org.antlr.v4.runtime.CommonTokenStream
+import org.antlr.v4.runtime.tree.ParseTree
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+internal class ASTNodesKtTest {
+
+    private var countNode: Int = 0
+    private var countChild: Int = 0
+    private var countSibling: Int = 0
+
+    @BeforeEach
+    fun setUp() {
+        countNode = 0
+        countChild = 0
+        countSibling = 0
+    }
+
+    private fun createParserTree(inputAntlrString: String): ParseTree {
+        val lexer = TreepatLexer(CharStreams.fromString(inputAntlrString))
+        val tokenStream = CommonTokenStream(lexer)
+        val parser = TreepatParser(tokenStream)
+        return parser.subtree()
+    }
+
+    private fun countNodes(node: ASTNode) {
+        when (node) {
+            is Node -> countNode++
+            is Child -> {
+                countNodes(node.father)
+                countNodes(node.child)
+                countChild++
+            }
+            is Sibling -> {
+                node.siblings.map { countNodes(it) }
```

Comment:
We are not actually mapping the elements of `node.siblings` to anything so we can use `.forEach` instead. 
```suggestion
                node.siblings.forEach { countNodes(it) }
```